camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/EmptySoftwareProcessYamlTest.java

```diff
@@ -60,6 +62,31 @@ public void testProvisioningProperties() throws Exception {
     }
 
     @Test(groups="Integration")
+    public void testProvisioningPropertiesOverrideValues() throws Exception {
+        Entity app = createAndStartApplication(
+                "location:",
+                "  localhost:",
+                "    templateOptions:",
+                "      testKey: testValue",
```

Comment:
I'd include in the test an assertion about what takes precedence. (and you can turn off all attempts to ssh to the machine). For example, have:

    location:
      useJcloudsSshInit: false
      waitForSshable: false
      minRam: 1G
      minCores: 1
      templateOptions:
        locKey1: locVal1
        locKey2: locVal2
    services:
    - type: EmptySoftwareProcess
      brooklyn.config:
        onbox.base.dir.skipResolution: true
        sshMonitoring.enabled: false
        provisioning.properties:
          minRam: 2G
          templateOptions:
            locKey1: overriddenLocVal1
            entKey1: entVal1